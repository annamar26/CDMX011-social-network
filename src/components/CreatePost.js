import { fbFunctions } from '../firebaseClient.js';
import { onNavigate } from '../routes.js';

const CreatePost = () => {
  const conteiner = document.createElement('div');
  const html = `  <nav>
    <div>
    <img class='logo' src='./icons&img/logo.png'></div>
    <div id='profile' class='icons img-profile'>
    </div>
       </nav>
       <div class='create-post'>
    <form id='postForm'>
    <h3>Crear publicación</h3>
    <input id='autor' type=text placeholder="Autor del libro" required>
    <input id='title' type=text placeholder="Titulo del libro" required>
    <textarea id='content-input' type=text placeholder="Escribe aquí tu reseña. Cuenta tu experiencia con la lectura del libro que desees" maxlength="1200" required></textarea>
    <button id='publicar' class='button' type=submit>Publicar</button>
    </form>
    </div>`;

  conteiner.innerHTML = html;
  fbFunctions.setWelcome();
  fbFunctions.comprobar();
  const postCreated = conteiner.querySelector('#postForm');
  postCreated.addEventListener('submit', (e) => {
    e.preventDefault();
    const bookAutor = conteiner.querySelector('#autor').value;
    const bookTitle = conteiner.querySelector('#title').value;
    const postContent = conteiner.querySelector('#content-input').value;

    fbFunctions.createPost(bookAutor, bookTitle, postContent)
      .then(() => {
        console.log('Document successfully written!');
        fbFunctions.getPosts();
        onNavigate('/timeline');
      })
      .catch((error) => {
        console.error('Error writing document: ', error);
      });
  });
  return conteiner;
};

export default CreatePost;
