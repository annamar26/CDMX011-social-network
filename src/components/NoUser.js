import { onNavigate } from "../routes.js";

const noUser = () => {
  const conteiner = document.createElement('div');

  const html = `<nav>
      <img class='logo' src='./icons&img/logo.png'>
     
      
      </nav>
      <div class='noUser-view signUp-container'>
      <h2>Debes iniciar sesion para ver el contenido de la pagina</h2>
      <br>
      <br>
      <input id='loginSubmit' class='button' type=button value='Inicia SesiÃ³n'>
      </div>`;

  conteiner.innerHTML = html;
  const volver = conteiner.querySelector('#loginSubmit');
  volver.addEventListener('click', ()=>{
    onNavigate('/login');
  });

  return conteiner;
};

export default noUser;
