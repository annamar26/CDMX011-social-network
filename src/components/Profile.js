/* eslint-disable max-len */
/* eslint-disable no-nested-ternary */
import { fbFunctions } from '../firebaseClient.js';
import { onNavigate } from '../routes.js';

export const Profile = () => {
  const container = document.createElement('div');
  const html = `<nav>
    <div>
    <img class=' logo logo1' src='./icons&img/logo.png'> </div>
    <div id='profile' class='icons img-profile'>
   <img class='logout'src='./icons&img/logout.svg'>
    </div></nav>
  
    <div class='profile-conteiner'>
    <div class='profile'>
    <h2 class='instruccion'>Configura tu perfil</h2>
  <img id='profileImage' src=''>
  <form class='profile-form'> 
 
  <label for='Nombre'>Nombre del perfil</label>
  <input id='Nombre' value='' type=text placeholder='Pablito Perez'>
  <label for='Email'>Email</label>
  <input id='Email' value='' type=text >
  <input id='photo' value='Selecciona tu foto de perfil'  type=file acept='image/*'>
 
  <button id='publicar' class='button' type=submit>Guardar Cambios</button></form>
 </div>
   <div id='postUserCointainer' class='profile-timeline posts-container'></div>
   <div id='footer-conteiner'>
  <footer id='footer'>
  <img class='icon' id='createPost' src='./icons&img/createicon.svg'>  </div>
    </footer>
    </div>`;

  container.innerHTML = html;
  fbFunctions.getCurrentUserPosts();
  fbFunctions.comprobar();
  fbFunctions.setDataProfile();

  const logout = container.querySelector('.logout');
  logout.addEventListener('click', () => {
    fbFunctions.userLogout().then(() => {
      console.log('logout exitoso');
    });
  });

  container.querySelector('#createPost').addEventListener('click', (e) => {
    e.preventDefault();
    onNavigate('/createpost');
  });

  const data = container.querySelector('.profile-form');
  data.addEventListener('submit', (e) => {
    e.preventDefault();
    const name = document.getElementById('Nombre').value;
    const email = document.getElementById('Email').value;
    const photo = document.getElementById('photo').files[0];

    fbFunctions.updateStringsProfile(name, email)
      .then(() => {
        console.log('Perfil editado');
      })
      .catch((error) => {
        console.error('Error writing document: ', error);
      });

    fbFunctions.pushNewPhoto(photo, photo.name)
      .then(() => {
        console.log('Perfil editado');
        fbFunctions.pullNewPhoto(photo.name).then((url) => {
          fbFunctions.updatePhotoProfile(url).then(() => {
            fbFunctions.updateUserDoc(name, email, url);
            onNavigate('/profile');
          });
        });
      })
      .catch((error) => {
        console.error('Error writing document: ', error);
      });
  });

  return container;
};
