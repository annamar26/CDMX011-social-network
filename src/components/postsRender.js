import { fbFunctions, router } from '../index.js';
import timeline from './timeline.js';

export const postConteiner = {

  setupPosts: (data) => {
    const postContent = document.querySelector('.posts-container');
    let html = '';
    const currentUser = fbFunctions.getUser();

    let currentidPost = '';
    if (data.length) {
      return data.forEach((doc) => {
        const publicaciones = doc.data();
        const date = publicaciones.fecha.toDate().toLocaleString();

        const li = `
         <div class='post-div'>
         <div class='post-header'>
         <div class='div-user'><img class='photo-user user-post' src='${publicaciones.userphoto ? publicaciones.userphoto : './icons&img/profileicon.svg'}'><p class='user'>${publicaciones.user}</p></div>
         <div class='fecha'>
         <p >${date}</p> </div>
         </div>
         <div class='post-autor'>
          <h2>${publicaciones.title} </h2> <h4> de ${publicaciones.autor}</h4></div>
          <div class='content'>
          <p class='content-text'>${publicaciones.text}</p></div>
         
          <div class='like-buttons post-header '>
          <img class='post-icon' src='' data-id='${doc.id}' data-likes='${publicaciones.likes}' id='like-${doc.id}'>
          <a class='tooltip' title='${publicaciones.likes}'>${publicaciones.likes.length} likes </a></div>
          
          <div class='delete-buttons post-header '>
          <img src='' class='delete' data-uid=${publicaciones.uid} data-id='${doc.id}' id='delete-${doc.id}'>
          <img src='' class='edit' data-uid=${publicaciones.uid} data-id='${doc.id}' id='edit-${doc.id}'>
          </div>
          
          </div>
          `;

        html += li;
        postContent.innerHTML = html;

        const likes = document.querySelectorAll('.post-icon');

        for (const like of likes) {
          if (like.dataset.likes.includes(currentUser.email) === true) {
            like.src = './icons&img/likedicon.svg';
          } else {
            like.src = './icons&img/likeicon.svg';
          }
          like.addEventListener('click', () => {
            if (like.dataset.likes.includes(currentUser.email) === true) {
              currentidPost = like.dataset.id;

              fbFunctions.deleteLike(currentidPost, currentUser.email).then(() => {
                fbFunctions.getPosts();

                console.log(currentidPost);
                console.log('dislike');
              })
                .catch((error) => {
                  console.error('Error writing like: ', error);
                });
            } else {
              currentidPost = like.dataset.id;

              fbFunctions.createLike(currentidPost, currentUser.email).then(() => {
                fbFunctions.getPosts();
                console.log(currentidPost);
                console.log('Like');
              })
                .catch((error) => {
                  console.error('Error writing like: ', error);
                });
            }
          });
        }

        const buttons = document.querySelectorAll('.delete');
        for (const button of buttons) {
          if (button.dataset.uid === currentUser.uid) {
            button.src = './icons&img/deleteicon.svg';

            button.addEventListener('click', () => {
              currentidPost = button.dataset.id;
              const result = confirm('¿Estás seguro de que quieres eliminar esta publicación?');
              if (result === true) {
                fbFunctions.deletePost(currentidPost).then(() => {
                  console.log('Document successfully deleted!');
                  fbFunctions.getPosts();
                })
                  .catch((error) => {
                    console.error('Error removing document: ', error);
                  });
              }
            });
          } else {
            button.src = '';
          }
        }

        const edits = document.querySelectorAll('.edit');
        for (const edit of edits) {
          if (edit.dataset.uid === currentUser.uid) {
            edit.src = './icons&img/editicon.svg';

            edit.addEventListener('click', () => {
              currentidPost = edit.dataset.id;
              const post = fbFunctions.getPostToEdit(currentidPost)
                .then((doc) => {
                  doc.data();

                  html = `  <nav>
                  <div>
                  <img class='logo' src='./icons&img/logo.png'></div>
                  <div id='profile' class='icons img-profile'>
                  </div>
                     </nav>
                     <div class='edit-post create-post'>
                  <form id='postForm'>
                  <h3>Edita tu publicación</h3>
                  <input id='autor' value='${doc.data().autor}' type=text placeholder="Autor del libro" required>
                  <input id='title' value='${doc.data().title}' type=text placeholder="Titulo del libro" required>
                  <textarea id='content-input'  type=text placeholder="Escribe aqií tu reseña" maxlength="1200" required>${doc.data().text}</textarea>
                  <button id='publicar' class='button' type=submit>Guardar Cambios</button>
                  </form>
                  </div>`;

                  const rootDiv = document.querySelector('#root');
                  rootDiv.innerHTML = html;
                  fbFunctions.setWelcome();
                  const autor = rootDiv.querySelector('#autor');
                  autor.setAttribute = ('value', post.autor);
                  const editedPost = document.querySelector('#postForm');
                  editedPost.addEventListener('submit', (e) => {
                    e.preventDefault();
                    const bookAutor = document.getElementById('autor').value;
                    const bookTitle = document.getElementById('title').value;
                    const postContent = document.getElementById('content-input').value;

                    fbFunctions.updatePost(edit.dataset.id, bookAutor, bookTitle, postContent)
                      .then(() => {
                        console.log('Document successfully updated!');
                        timeline.template();
                      })
                      .catch((error) => {
                        console.error('Error updated document: ', error);
                      });
                  });
                  return rootDiv;
                })

                .catch((error) => {
                  console.log('Error getting document:', error);
                });
            });
          } else {
            edit.src = '';
          }
        }
      });
    }

    html = '';
    postContent.innerHTML = html;
    return postContent;
  },

};
