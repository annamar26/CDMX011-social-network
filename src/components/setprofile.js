/* eslint-disable max-len */
/* eslint-disable no-nested-ternary */
import { fbFunctions } from '../index.js';

const setProfile = {
  template() {
    const html = `<nav>
    <div>
    <img class='logo1 logo' src='./icons&img/logo.png'> </div>
    <div id='profile' class='icons img-profile'>
   <img class='logout'src='./icons&img/logout.svg'>
    </div></nav>
  
    <div class='profile-conteiner'>
    <div class='profile'>
  <img id='profileImage' src=''>
  <form class='profile-form'> 
 
  <input id='Nombre' value='' type=text >
  <input id='Email' value='' type=text >
  <input id='photo' value='Selecciona tu foto de perfil'  type=file acept='image/*'>
 
  <button id='publicar' class='button' type=submit>Guardar Cambios</button></form>
 </div>
   <div id='postUserCointainer' class='profile-timeline posts-container'></div>
   <div id='footer-conteiner'>
  <footer id='footer'>
  <img class='icon' id='createPost' src='./icons&img/createicon.svg'>  </div>
    </footer>
    </div>`;

    const rootDiv = document.querySelector('#root');
    rootDiv.innerHTML = html;
    fbFunctions.getCurrentUserPosts();
    fbFunctions.comprobar();
    fbFunctions.setDataProfile();

    const logout = document.querySelector('.logout');
    logout.addEventListener('click', () => {
      fbFunctions.userLogout().then(() => {
        console.log('logout exitoso');
      });
    });

    const data = document.querySelector('.profile-form');
    data.addEventListener('submit', (e) => {
      e.preventDefault();
      const name = document.getElementById('Nombre').value;
      const email = document.getElementById('Email').value;
      const photo = document.getElementById('photo').files[0];

      fbFunctions.updateStringsProfile(name, email)
        .then(() => {
          console.log('Perfil editado');
        })
        .catch((error) => {
          console.error('Error writing document: ', error);
        });

      fbFunctions.pushNewPhoto(photo, photo.name)
        .then(() => {
          console.log('Perfil editado');
          fbFunctions.pullNewPhoto(photo.name).then((url) => {
            fbFunctions.updatePhotoProfile(url).then(() => {
              setProfile.template();
            });
          });
        })
        .catch((error) => {
          console.error('Error writing document: ', error);
        });
    });

    return rootDiv;
  },
  setProfile() {
    const photo = document.getElementById('profileImage');
    const name = document.getElementById('Nombre');
    const email = document.getElementById('Email');
    /*   const postContent = document.getElementById('content-input').value; */
    photo.src = fbFunctions.getUser() ? fbFunctions.getUser().photoURL === null ? '../icons&img/profileicon.svg' : fbFunctions.getUser().photoURL : null;
    email.value = fbFunctions.getUser() ? fbFunctions.getUser().email : null;
    name.value = fbFunctions.getUser() ? fbFunctions.getUser().displayName : null;
  },

};
export default setProfile;
