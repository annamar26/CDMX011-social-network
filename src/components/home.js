/* eslint-disable import/no-cycle */
import { router } from '../index.js';
import signup from './signup.js';
import login from './login.js';

export const home = {
  template: () => {
    const html = `<nav>
    <img class='logo' src='./icons&img/logo.png'>
    <img class='menu' src='./icons&img/menu.svg'/>
    
    </nav>
    <main>
<div class='main-container'>
    <div class='welcome-container'>
   
            <h1>La comunidad más grande de lectores</h1>
            
      
    </div>
    
    <div class='secondary-container'>
    <input id='loginButton' type='button' value='Inicia sesión'>
    <p>¿No tienes una cuenta?</p>
    <a id='signUpA'>Regístrate</a>
    <p id='ms'></p>
    </div>
  
</div>


</main>

`;

    document.querySelector('#root').innerHTML = html;

    document.querySelector('#signUpA').addEventListener('click', (e) => {
      e.preventDefault();
      router.onNavigate('/signup');
      signup.template();
    });

    document.querySelector('#loginButton').addEventListener('click', (e) => {
      e.preventDefault();
      router.onNavigate('/login');
      login.template();
    });
  },
};
