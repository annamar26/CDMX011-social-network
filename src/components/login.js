/* eslint-disable import/no-cycle */
import { fbFunctions } from '../index.js';
import { router } from '../index.js';
import { home } from './home.js';
import timeline from './timeline.js';

const login = {

  template() {
    const html = `<nav>
    <img class='logo' src='./icons&img/logo.png'>
   
    
    </nav>
    <div class='signUp-container'>
    <h2>Inicia sesi칩n</h2>
   
    <form class=form id=loginForm>

    <label for='lgEmail'>Email</label>
        <input id='lgEmail' type='email' required>
        <label for='lgPassword'>Contrase침a</label>
        <input id='lgPassword' type='password' required><img id='seeLps' src='./icons&img/ver.svg'>
         <p class='warning' id='errorMessage1'></p>
         <input id='loginSubmit' type='submit' value='Iniciar sesi칩n'>
      
    </form>
    <img class='div' src='./icons&img/div.svg'>
    <button class='button' id='googleL'><img class='gicon' src='./icons&img/googleicon.svg'>Inicia sesi칩n con Google</button>
    <img id='return' class='return'src='./icons&img/return.svg'>
    </div> `;

    const rootDiv = document.querySelector('#root');
    rootDiv.innerHTML = html;

    document.getElementById('googleL').addEventListener('click', (e) => {
      e.preventDefault();
      fbFunctions.googleUserSignUp(e)
        .then(() => {
          console.log('Google singin');
          timeline.template();
          router.onNavigate('/timeline');
          fbFunctions.getPosts();
        })
        .catch((error) => { document.getElementById('errorMesagge1').innerHTML = error; });
    });

    document.getElementById('return').addEventListener('click', (e) => {
      e.preventDefault();
      home.template();
      router.onNavigate('/');
    });

    document.querySelector('#seeLps').addEventListener('click', (e) => {
      e.preventDefault();
      const tipo = document.getElementById('lgPassword');
      if (tipo.type === 'password') {
        tipo.type = 'text';
      } else {
        tipo.type = 'password';
      }
    });

    const loginData = document.querySelector('#loginForm');
    loginData.addEventListener('submit', (e) => {
      e.preventDefault();
      const loginEmail = document.querySelector('#lgEmail').value;
      const loginPassword = document.querySelector('#lgPassword').value;
      fbFunctions.userLogin(loginEmail, loginPassword).then(() => {
        console.log('Login exitoso');
        timeline.template();
        router.onNavigate('/timeline');
        fbFunctions.getPosts();
      })
        .catch((error) => { document.getElementById('errorMessage1').innerHTML = error; });
    });
    return rootDiv;
  },
};
export default login;
