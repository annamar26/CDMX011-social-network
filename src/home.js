import { router } from './lib/index.js';
import signup from './signup.js';
import login from './login.js';

export const home = {
  template: () => {
    const html = `<nav>
    <img class='logo' src='logo.png'>
    <img class='menu' src='menu.svg'/>
    
    </nav>
    <main>
<div class='main-container'>
    <div class='welcome-container'>
   
            <h1>La comunidad más grande de lectores</h1>
            
      
    </div>
    
    <div class='secondary-container'>
    <input id='loginButton' type='button' value='Inicia sesión'>
    <p>¿No tienes una cuenta?</p>
    <a id='signUpA'>Regístrate</a>
    <p id='ms'></p>
    </div>
  
</div>


</main>
<footer></footer>
`;

    document.querySelector('#root').innerHTML = html;
    router.onNavigate('/');
    document.querySelector('#signUpA').addEventListener('click', () => {
      router.onNavigate('/signup');
      signup.template();
    });

    document.querySelector('#loginButton').addEventListener('click', () => {
      router.onNavigate('/login');
      login.template();
    });
  },
};
