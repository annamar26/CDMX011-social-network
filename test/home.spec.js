// importamos la funcion que vamos a testear
import { home.template() } from '../src/components/home'

describe('home.template(', () => {
  it('should render home page', () => {
    expect().toBe('function');
  });
});
