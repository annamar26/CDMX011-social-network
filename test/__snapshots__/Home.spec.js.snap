// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render Home page 1`] = `
<div
  id="root"
>
  <div>
    <nav>
      
    
      <img
        class="logo"
        src="./icons&img/logo.png"
      />
      
   
    
    </nav>
    

    
    <main>
      
    
      <div
        class="main-container"
      >
        

    
        <div
          class="welcome-container"
        >
          
    
          <h1>
            La comunidad más grande de lectores
          </h1>
          
    
        </div>
        
    
    
        <div
          class="secondary-container"
        >
          
    
          <input
            class="login-button"
            id="loginButton"
            type="button"
            value="Inicia sesión"
          />
          
    
          <p>
            ¿No tienes una cuenta?
          </p>
          
    
          <a
            id="signUpA"
          >
            Regístrate
          </a>
          
    
          <p
            id="ms"
          />
          
    
        </div>
        
  
    
      </div>
      
    
    </main>
  </div>
</div>
`;
